### ISSUE TITLE
Unit test data looks wrong

### ISSUE THREAD

**Issue by msound** [coreos/go-semver#23]
In unit test `TestBumpMajor` (and in other places), the test data being used in [Line 178](https://github.com/coreos/go-semver/blob/5e3acbb5668c4c3deb4842615c4098eb61fb6b1e/semver/semver_test.go#L178) is:
```
1.0.0+build.1-alpha.1
```
Prerelease should come first and must have a prefix of `-`
Metadata should come in the end and must have a prefix of `+`.
This test data is the other way around.

Let's see what happens when we use this test data in an example:
```
package main

import "fmt"
import "github.com/coreos/go-semver/semver"

func main() {
    v := semver.New("1.0.0+build.1-alpha.1")
    fmt.Printf("PreRelease: %s\n", v.PreRelease)
    fmt.Printf("Metadata: %s\n", v.Metadata)
}
```

And the output is: 
```
PreRelease: 
Metadata: build.1-alpha.1
```

Therefore the testing in [Line 180](https://github.com/coreos/go-semver/blob/5e3acbb5668c4c3deb4842615c4098eb61fb6b1e/semver/semver_test.go#L180) is not really effective, considering that PreRelease was `""` to begin with.

I propose that the test data be changed to:
```
1.0.0-alpha.1+build.1
```

If you agree, let me know and I will work on a pull request.

**Comment by lucab** [comment]
> Therefore the testing in Line 180 is not really effective, considering that PreRelease was "" to begin with.

You are right, this is not a major issue with the test but your proposed changes would increase coverage.

> If you agree, let me know and I will work on a pull request.

Please proceed with it. It would also be nice to have a dedicated test case in `TestMust` to test against PreRelease and Metadata mixups.
